codec/README.md

```diff
@@ -0,0 +1,58 @@
+# Codec
+
+**This is currently a DRAFT project.**
+
+## Goal
+
+Describe an algorithm that can:
+
+- Detect encoded substrings in a string 
+- Replace encoded substrings with their decoded values
+- Decode encoded substrings that overlap or are contained in previously decoded substrings 
+- Map a decoded substring's position to its encoded position the original string 
+- Be performant enough to enable by default in most scenarios
+
+Describe approaches like the following for scanning the resulting string(s) for secrets:
+
+- Only scanning the fully decoded string 
+- Scanning the string at each layer of decoding 
+  - Without double reporting findings
+  - Why this might be desirable
+
+Describe it in a way that it can be integrated into different scanners.
+
+Because things like:
+[intentional obfuscation by malware](https://thehackernews.com/2025/11/second-sha1-hulud-wave-affects-25000.html), 
+[tools requiring encoded values](https://kubernetes.io/docs/concepts/configuration/secret/),
+[formatting codes](https://en.wikipedia.org/wiki/ANSI_escape_code),
+[partial encodings](https://en.wikipedia.org/wiki/Percent-encoding),
+etc make pure regex secret detection difficult or impossible.
```

Comment:
This is a good source to reference for that:
https://semgrep.dev/blog/2025/secrets-story-and-prefixed-secrets/